From: Carlos Garnacho <carlosg@gnome.org>
Date: Tue, 15 Aug 2023 17:13:32 +0200
Subject: clutter: Add boolean return value to
 clutter_stage_get_device_coords()

The device/sequence may not currently have a set of coordinates to return.
We correctly leave the out values uninitialized, but don't tell the upper
layers in any way.

(cherry-picked from commit 88af55b8576e3074638619b1a3ef9f29fac8c9a0)

Origin: upstream, 44.6, commit:497f70557ccd1aaf7ee53e9fdaa7d8ba6e15f082
---
 clutter/clutter/clutter-mutter.h |  8 ++++----
 clutter/clutter/clutter-stage.c  | 13 +++++++++----
 2 files changed, 13 insertions(+), 8 deletions(-)

diff --git a/clutter/clutter/clutter-mutter.h b/clutter/clutter/clutter-mutter.h
index 3250eb5..6c96949 100644
--- a/clutter/clutter/clutter-mutter.h
+++ b/clutter/clutter/clutter-mutter.h
@@ -119,10 +119,10 @@ void clutter_stage_update_device (ClutterStage         *stage,
                                   gboolean              emit_crossing);
 
 CLUTTER_EXPORT
-void clutter_stage_get_device_coords (ClutterStage         *stage,
-                                      ClutterInputDevice   *device,
-                                      ClutterEventSequence *sequence,
-                                      graphene_point_t     *coords);
+gboolean clutter_stage_get_device_coords (ClutterStage         *stage,
+                                          ClutterInputDevice   *device,
+                                          ClutterEventSequence *sequence,
+                                          graphene_point_t     *coords);
 CLUTTER_EXPORT
 void clutter_stage_repick_device (ClutterStage       *stage,
                                   ClutterInputDevice *device);
diff --git a/clutter/clutter/clutter-stage.c b/clutter/clutter/clutter-stage.c
index 74a23b6..70bc00b 100644
--- a/clutter/clutter/clutter-stage.c
+++ b/clutter/clutter/clutter-stage.c
@@ -3386,7 +3386,7 @@ clutter_stage_get_device_actor (ClutterStage         *stage,
 /**
  * clutter_stage_get_device_coords: (skip):
  */
-void
+gboolean
 clutter_stage_get_device_coords (ClutterStage         *stage,
                                  ClutterInputDevice   *device,
                                  ClutterEventSequence *sequence,
@@ -3395,16 +3395,21 @@ clutter_stage_get_device_coords (ClutterStage         *stage,
   ClutterStagePrivate *priv = stage->priv;
   PointerDeviceEntry *entry = NULL;
 
-  g_return_if_fail (CLUTTER_IS_STAGE (stage));
-  g_return_if_fail (device != NULL);
+  g_return_val_if_fail (CLUTTER_IS_STAGE (stage), FALSE);
+  g_return_val_if_fail (device != NULL, FALSE);
 
   if (sequence != NULL)
     entry = g_hash_table_lookup (priv->touch_sequences, sequence);
   else
     entry = g_hash_table_lookup (priv->pointer_devices, device);
 
-  if (entry && coords)
+  if (!entry)
+    return FALSE;
+
+  if (coords)
     *coords = entry->coords;
+
+  return TRUE;
 }
 
 static void
